submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/PersistenceContextImpl.kt

```diff
@@ -38,15 +39,17 @@ open class PersistenceContextImpl(
     }
 
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
-    override fun saveSubmission(submission: ExtSubmission, submitter: String, submitterId: Long): ExtSubmission {
-        return lockExecutor.executeLocking(submission.accNo) {
-            subRepository.expireActiveVersions(submission.accNo)
-            deleteSubmissionDrafts(submitterId, submission.accNo)
+    override fun saveSubmission(saveRequest: SaveRequest): ExtSubmission {
+        val (submission, submitter, mode) = saveRequest
+        return lockExecutor.executeLocking(submission.accNo) { saveSubmission(submission, submitter, mode) }
+    }
 
-            val newSubmission = saveNewVersion(submission, submitter)
-            filePersistenceService.persistSubmissionFiles(submission)
-            newSubmission
-        }
+    private fun saveSubmission(submission: ExtSubmission, submitter: User, mode: FileMode): ExtSubmission {
```

Comment:
Not sure why we need this method. I think it looks clear just executing this in the `executeLocking` closure